chore: update changelog (#1981)
berops/claudie: docs/CHANGELOG/changelog-0.10.x.md

# Claudie `v0.10`

!!! warning "Several major internal changes have been made to Claudie. While the `v0.10.0` version should be backward compatible with `v0.9.16`, it was not possible to test all possible cluster configuration scenarios. We advise creating backups before upgrading to v0.10.0 and later versions."

!!! note "only v0.10.0 should be backwards compatible with the v0.9.16 version any other v0.10.x version will not have this guarantee"

## Most notable changes (TL;DR)

- After deploying claudie version `v0.10.0` the reconciliation loop will be initiated after the first `kubectl apply -f <your-input-manifest>` 

>   that has a change in the desired state compared to the last applied version.

Reviewer: ```suggestion
  that changes how the cluster desired state is treated.
```
?

Not sure what you tried to say here. Perhaps you can reword this part of the sentence?

Author: Exactly what I have written, The reconciliation loop will be trigger by a `kubectl apply -f` that updates the InputManifest, i.e. is different then the last applied version

- The identification and scheduling of tasks has been overhauled. Claudie now has an initial version of a reconciliation loop. In the v0.9.x versions of Claudie, whenever a change was detected after running `kubectl apply -f <your-input-manifest>` Claudie stopped and did not continue to health check or fix the error, even if the error was simply a network inconvenience, upon either a failure or success of building that change. As of now, with the reconciliation loop, every `kubectl apply -f <your-input-manifest>` will explicitly state the desired state of your clusters, and Claudie will try endlessly to reach that desired state. This means that, in the event of any errors, changes will be reverted and then reapplied, along with health checking, which helps identify potential misconfigurations or infrastructure issues. Claudie will then try to auto-repair these issues, if possible. The goal is to further improve the reconciliation loop with each release.

> - DynamoDB was removed in favor of native locking supported by newer versions of OpenTofu which ship with claudie `v0.10.x`

Reviewer 2: ```suggestion
- DynamoDB was removed in favor of native locking supported by newer versions of OpenTofu which ship with Claudie `v0.10.x`
```
